### ISSUE TITLE
[Translate] Russian language

### ISSUE THREAD

**Issue by MHStrider** [CorsixTH/CorsixTH#1187]
###
[Notation] ==> I'm used Google Translate + school experience, sorry for my English.
###

Hello! In some part of the game interface is not translated, translated or not quite right. It seems to be an update out not often, but do not have full support for the language.
How can I help to improve the translation? I am a "noob" on github. And I not really know what and where.
Still I can offer translation in the Ukrainian language, but I think he is not particularly in demand.

Thank you for your attention and look forward to your reply!

**Comment by Alberth289346** [comment]
https://github.com/CorsixTH/CorsixTH/wiki/Localization should explain most of the stuff you want to know

As for giving us results, we accept
- proper PRs (pull requests) if you know how to make them, or
- just edit the language file, and attach the modified language file to this issue, and we'll merge it.

Thanks!

**Comment by StriderxD** [comment]
**Alberth289346**, thanks for the reply.
I'm probably going to use the method "pull requests", so probably easier.
Thank you all ( the one who came up with and the people who help to refine ) for such a modification.
Happy New Year everyone! Happy holiday!

**Comment by Alberth289346** [comment]
PR would be great!

Some things to watch out for, in that case:
- No TABs, only use spaces.
  Different people have different ideas about length of TABs. To avoid problems, we only use spaces. To be precise 2 spaces at each indent level.
  It's simplest if your editor can convert TABs to spaces.

- No trailing whitespace.
  In particular, many editors make a mess at empty lines.
  Simplest for checking is to search for a space at the end of a line. If you do this search after you replaced TABs to spaces, you can remove trailing TABs at the same time.

**Comment by MHStrider** [comment]
Well, since all Lua files can be edited and viewed locally. That, after all going to do on your laptop, check in the game after already sent here.

I use Notepad ++, I think you know about it, a good tool.

And some do not understand about "No trailing whitespace" If possible, show an example. Google it does not know how to translate and does not know how. Too complicated the language - Russian :)

**Comment by MHStrider** [comment]
Few reviewed the file, and I do not like here for example the structure of the russian.lua file. It seems that just after the update added the option you want and its translation. It will be necessary to reduce the amount of code a bit. Even it will be interesting, as it himself throw a challenge.
Note: russian.lua <== 249 kb and 3426 lines

Original ( or reference / sample / standart ) structure of the english.lua:
====387-399 lines
customise_window = {
  caption = "Custom Settings",
  option_on = "On",
  option_off = "Off",
  back = "Back",
  movies = "Global Movie Control",
  intro = "Play Intro Movie",
  paused = "Build on Paused",
  volume = "Volume down hotkey",
  aliens = "Alien Patients",
  fractured_bones = "Fractured Bones",
  average_contents = "Average contents",
}
====

Structure of the russian.lua:
====3392-3402 lines
customise_window.average_contents = "Среднее содержимое"
customise_window.option_on = "Вкл"
customise_window.paused = "Строить во время паузы"
customise_window.option_off = "Выкл"
customise_window.intro = "Проигрывать заставку"
customise_window.caption = "Пользовательские настройки"
customise_window.back = "Назад"
customise_window.movies = "Управление роликами"
customise_window.volume = "Горячая клавиша убавления громкости"
customise_window.aliens = "Инопланетные пациенты"
====

**Comment by MHStrider** [comment]
I created a repository, to learn to work with github and can operate remotely.

https://github.com/MHStrider/LangTH

**Comment by MHStrider** [comment]
Why in the file russian.lua is the value of:
adviser.tutoruial.build_reception
And in "fayl.uchu" no, where he takes this value?

**Comment by Alberth289346** [comment]
- Notepad++ should work. Please make sure you use "UTF-8" (or "Unicode") when saving the file.

- "Trailing whitespace" are spaces or TAB characters _after_ the last letter of a line.

Since whitespace is hard to see, I'll given an example with "trailing %"
```` text
%%some text%%%%
^^         ^^^^^ Trailing %, these should not be here.
Leading %, and that's ok
````

> Few reviewed the file, and I do not like here for example the structure of the russian.lua file. It seems that just after the update added the option you want and its translation. It will be necessary to reduce the amount of code a bit. Even it will be interesting, as it himself throw a challenge.

Both forms are the same in Lua, you are free to change the lines, but do 1 thing at a time. Either improve translation, or modify structure, but not together.

If you do them together it becomes very hard to understand what is being modified.

- If you have not done already, you can fork the CorsixTH repository to your own account, and work in there. (Use the "Fork" button at the top-right of this page)
https://help.github.com/articles/fork-a-repo/  gives more details

- adviser.tutorial.build_reception (and many others):
The game uses the English strings of the original Theme Hospital disk. The "english.lua" file only contains new strings (not in the original version), and a few fixed strings, I think.

Since the original Theme Hospital disk does not have Russian strings, you must provide a Russian translation for all original strings (such as adviser.tutorial.build_reception), and you must provide a Russian translation for all new and fixed strings.
For this reason, many of the language files are very large.

**Comment by lewri** [comment]
Resolved by #2065
